lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -389,10 +512,42 @@ def _list_backups(self) -> Dict[int, str]:
         }
 
     def is_backup_in_progress(self, backup_id=None) -> bool:
-        """Returns True if backup is in progress, False otherwise."""
-        return self._query_backup_status(backup_id) in [
+        """Returns True if backup is in progress, False otherwise.
+
+        First, check if we have a marker of a running backup. If yes, then we check for both the
+        saved backup_id in peer relation AND backup_id, if provided. If neither are in progress,
+        return False and clean up the marker from the relation.
+        """
+        rel = self.model.get_relation(PeerRelationName)
+        if rel:
+            # If relation is present, then check the peer relation data
+            peer_relation_backup_id = rel.data[self.charm.app].get("backup_in_progress", None)
+            if peer_relation_backup_id and self._query_backup_status(backup_id) in [
+                BackupServiceState.SNAPSHOT_IN_PROGRESS
+            ]:
+                # We have a backup in progress
+                return True
+            else:
+                # Clean up status
+                if (
+                    self.charm.unit.is_leader()
+                    and "backup_in_progress" in rel.data[self.charm.app]
+                ):
+                    del rel.data[self.charm.app]["backup_in_progress"]
+        backup_id_in_progress = self._query_backup_status(backup_id) in [
             BackupServiceState.SNAPSHOT_IN_PROGRESS,
         ]
+        if backup_id_in_progress:
+            # We have a backup in progress
+            if (
+                peer_relation_backup_id
+                and peer_relation_backup_id != backup_id_in_progress
+                and self.charm.unit.is_leader()
+            ):
+                # We need to update our peer relation as the IDs were different
+                rel.data[self.charm.app]["backup_in_progress"] = str(backup_id_in_progress)
```

Comment:
this method shouldn't have to do this, given its name indicates that its sole focus is to check whether the backup is in progress, consider moving this to the caller side, or slightly rename the method